### DISCUSSION TITLE
Nextjs cache and cloudflare cache are not synced with getStaticProps revalidation

### DISCUSSION THREAD

**Discussion by eliezra236** [vercel/next.js#61485] [Help]
### Summary

I’m trying to test my nextjs application on digital ocean, I deployed it, and noticed that getStaticProps revalidation doesn’t work, I examined the cache headers and notice that the revalidation paramater is being asssigned to Cache-Control like follow: 

`Cache-Control: s-maxage=10800, stale-while-revalidate` 

Which is what I am for, to tell nextjs to regenerate the page every 3 hours to update data. But the issue is, nextjs and cloudflare are not synced. So now the status is: 
```
Cf-Cache-Status: HIT
X-Nextjs-Cache: STALE
```
Which means that for nextjs the page is expired, but for CF it’s not, so while nextjs might regenerate the page in the background it won’t show the new page until cloudflare cache will also expire.

Is there a way to fix it and sync them? Currently the application doesn’t have domain, so it’s all managed by digitalocean, but I am not sure if it's digitalocean issue or next ~ cloudflare issue that I can fix by configuration on either side.

### Additional information

```js
Using Next@12.3.4
```

**Comment by danielcalvolopez** [comment]
Have you considered updating to Next 14, and use fetch in server components instead of using getStaticProps? It is currently the recommended way to fetch data in Nextjs.

See doc here: https://nextjs.org/docs/app/api-reference/functions/fetch
